resources/assets/components/StatusButton/index.js

```diff
@@ -0,0 +1,18 @@
+import React from 'react';
+import './status-button.scss';
+// import { map } from 'lodash';
+
+
+class StatusButton extends React.Component {
+  getClass() {
+    return `button -${this.props.type}`;
+  }
+  // Ideal setStatus would be defined on this component and would set some sort of global state via redux
+  render() {
+    return (
+      <button className={this.getClass()} onClick={() => this.props.setStatus(this.props.type)}>{this.props.type}</button>
```

Comment:
This is totally subjective, but this looks like a great example of a functional component!

```js
import React from 'react';
import './status-button.scss';

export default (props) => (
  <button className={classnames('button', props.type} onClick={() => props.setStatus(props.type)}>
    {this.props.type}
  </button>
);
```